driver/flambda_backend_args.ml

```diff
@@ -537,6 +547,32 @@ struct
   ]
 end
 
+
+(* Build systems call 'ocamlopt' on a single file. When -seed is used for
```

Comment:
It sounds like we are rather confident
we will never use `files`, so why not get
rid of it? Possibly leaving a comment
explaining the assumption there is only
one file and why the code is as a result
different from the one in FDO.

Reply:
The difficulty in the current version (without the filenames) is that all compilation units start over with exactly the same seed. It would probably be better to start with a different seed, e.g., based on the input seed+file name of the current compilation unit. We can do it by re-initializing the random state at the beginning of each compilation unit (small change in asmgen, and moving around of some code). What do you think?

Reply:
You could indeed hash the name of the compilation
unit or/and the name of the function.

Reply:
good idea, I'll make it based on the function name - less random in a way but being independent of the order of functions in the file seems like a useful property for debugging (minimizing testcases and compare between versions).

Reply:
I don't think it will be independent of the
order of functions because of the stamp
at the end of the name.

Reply:
right.